src/libstd/process.rs

```diff
@@ -311,7 +312,8 @@ impl Read for ChildStdout {
 
     #[inline]
     unsafe fn initializer(&self) -> Initializer {
-        Initializer::nop()
+        // SAFETY: Read is guaranteed to work on uninitialized memory
```

Comment:
This is not quite accurate. In order for this to be safe, we basically need to guarantee that AnonPipe's `read` function is not going to read from the passed buffer. That's not a local property, nor is it an easy thing to check -- we have a bunch of AnonPipe impls most of which delegate to other struct themselves :/

It might be better to add a function to AnonPipe that returns an initializer and return that here, rather than making this assertion here. In the future this'll get replaced anyway once https://github.com/rust-lang/rfcs/pull/2930 is implemented, though, so that may not be worth it -- it might be best to

Reply:
So, what would be the correct comment here? It isn't safe, but that's how we do it currently? ;)

Reply:
It probably is safe and fine currently, but this comment makes it seem like that's "just true." Ideally, we would change the code to make reasoning about this local (e.g., add functions providing access to `Initializer` to each pipe and use those here). But that's a fairly large task. Alternatively, we can just write this comment being explicit about what has been checked. Something like "This is safe only if all implementations of AnonPipe correctly handle uninitialized data being passed into their read methods, which has not been confirmed to be the case" or something along those lines.